<a href="https://colab.research.google.com/github/galsaeedi/TensorFlow-Object-Detection-API-Multiple-Objects-GoogleColab/blob/master/OD.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
#trick to increase RAM usage on colab
a = []
while(1):
    a.append("1")

you need to **mount your google drive** into the cloud VM so that you can access its contents. just click mount Drive or run the following cell 



In [0]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


then we will create a symbolic link between '/content/gdrive/My\ Drive/' and '/mydrive.

In [0]:
#creates a symbolic link so that now the path /content/gdrive/My\ Drive/ is equal to /mydrive
#just to simplify things 
#if you run the above command dont run this :)
!ln -s /content/drive/My\ Drive/ /mydrive

In [0]:
!ls /mydrive

'Colab Notebooks'
 digit_data_set.zip
 digit_dataset.zip
 example.py
 extract_embeddings_knn.py
 extract_embeddings.py
 face_detection_confusion_matrix.py
 faster_rcnn_inception_v2_coco_2018_01_28
'faster_rcnn_inception_v2_coco_2018_01_28(1)'
 faster_rcnn_inception_v2_coco_2018_01_28.tar.gz
 faster_rcnn_inception_v2_coco_2018_01_28.zip
 faster_rcnn_inception_v2_pets.config
 files.JPG
 input.zip
 KNN_confusion_matrix.py
 label_transform_tom.pkl
 meter1201.jpg
 obj.names
 objxml.zip
 OCR
'poster_work_done2 copy.pdf'
 predict.py
 recognize_faces_image_knn.py
 recognize_faces_image_knn_video_cam.py
 recognize_faces_video_file_knn.py
 SVM_confusion_matrix.py
 Tensorflow1
 Tensorflow1.zip
 train_model.py
 weights.npz
 windows_v1.6.0.zip
 yolov3
 yolov3result.txt


#step 1: Enable GPU or TPU within your notebook

You need to enable GPU or TPU within your colab notebook 

1.   Click Edit at top left of your notebook
2.   Click Notebook Settings within dropdown
3.   Under 'Hardware Accelerator' select GPU or TPU and then Save it


#step 2: Install packages

In [0]:
!pip install --upgrade pip

     |████████████████████████████████| 1.5MB 3.4MB/s 
  Found existing installation: pip 19.3.1
    Uninstalling pip-19.3.1:
      Successfully uninstalled pip-19.3.1


In [0]:
pip install "numpy<1.17"

     |████████████████████████████████| 17.4 MB 140 kB/s 
ERROR: umap-learn 0.4.3 has requirement numpy>=1.17, but you'll have numpy 1.16.6 which is incompatible.
ERROR: datascience 0.10.6 has requirement folium==0.2.1, but you'll have folium 0.8.3 which is incompatible.
ERROR: albumentations 0.1.12 has requirement imgaug<0.2.7,>=0.2.5, but you'll have imgaug 0.2.9 which is incompatible.
  Attempting uninstall: numpy
    Found existing installation: numpy 1.18.4
    Uninstalling numpy-1.18.4:
      Successfully uninstalled numpy-1.18.4


**Specifying the TensorFlow version**

Running `import tensorflow` will import the default version (currently 2.x). You can use 1.x by running a cell with the `tensorflow_version` magic **before** you run `import tensorflow`.

In [0]:
%tensorflow_version 1.x

TensorFlow 1.x selected.


In [0]:
#!pip uninstall tensorflow

In [0]:
#install tensorflow with gpu version 1.15
!pip install tensorflow-gpu==1.15.2

     |████████████████████████████████| 411.0 MB 27 kB/s 
  Created wheel for gast: filename=gast-0.2.2-py3-none-any.whl size=7539 sha256=873aaf5233393a4a284c4800c7d9ea1164a28fa8b3ddee91c7a68de3fb445373
  Stored in directory: /root/.cache/pip/wheels/19/a7/b9/0740c7a3a7d1d348f04823339274b90de25fbcd217b2ee1fbe
Successfully built gast
  Attempting uninstall: gast
    Found existing installation: gast 0.3.3
    Uninstalling gast-0.3.3:
      Successfully uninstalled gast-0.3.3


In [0]:
import tensorflow as tf
tf.__version__

'1.15.2'

In [0]:
!pip install google 
!pip install grpc 
!pip install numpy 
!pip install pkg_resources 
!pip install six

ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement pkg_resources (from versions: none)
ERROR: No matching distribution found for pkg_resources


In [0]:
pip install tensorflow-object-detection-api

     |████████████████████████████████| 577 kB 3.5 MB/s 
     |████████████████████████████████| 54 kB 2.4 MB/s 
     |████████████████████████████████| 2.7 MB 18.6 MB/s 
  Created wheel for tensorflow-object-detection-api: filename=tensorflow_object_detection_api-0.1.1-py3-none-any.whl size=844513 sha256=197490bea981887cbb6dc49522bad1e2b4a5e4fe49559223a15259541352b3e8
  Stored in directory: /root/.cache/pip/wheels/1a/99/e7/ea0c585728781ffd2670ed324603ee4814c8a751830134be83
Successfully built tensorflow-object-detection-api
ERROR: readme-renderer 26.0 has requirement Pygments>=2.5.1, but you'll have pygments 2.1.3 which is incompatible.


In [0]:
!pip install protobuf
!pip install pillow
!pip install lxml
!pip install Cython
!pip install contextlib2
!pip install jupyter
!pip install matplotlib
!pip install pandas
!pip install opencv-python

In [0]:
pip install tf-nightly

     |████████████████████████████████| 525.1 MB 25 kB/s 
     |████████████████████████████████| 2.9 MB 76.6 MB/s 
     |████████████████████████████████| 459 kB 82.0 MB/s 
ERROR: tensorflow 1.15.2 has requirement gast==0.2.2, but you'll have gast 0.3.3 which is incompatible.
ERROR: tensorflow-gpu 1.15.2 has requirement gast==0.2.2, but you'll have gast 0.3.3 which is incompatible.
  Attempting uninstall: gast
    Found existing installation: gast 0.2.2
    Uninstalling gast-0.2.2:
      Successfully uninstalled gast-0.2.2


# step3 : clone tensorflow model repository from GitHub 


**before** cloning the repo create a folder and name it 'tensorflow1'. then clone the repo inside tensorflow1 folder


In [0]:
!mkdir tensorflow1

In [0]:
%cd tensorflow1/

/content/tensorflow1


clone the full TensorFlow object detection repository located at https://github.com/tensorflow/models

In [0]:
!git clone https://github.com/tensorflow/models

Cloning into 'models'...
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 36349 (delta 0), reused 1 (delta 0), pack-reused 36341
Receiving objects: 100% (36349/36349), 520.16 MiB | 49.00 MiB/s, done.
Resolving deltas: 100% (24222/24222), done.


In [0]:
%cd ..

/content


# step 4: Download the Faster-RCNN-Inception-V2-COCO model from TensorFlow's model zoo

TensorFlow provides several object detection models (pre-trained classifiers with specific neural network architectures) in its https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md



1.   Download Faster-RCNN-Inception-V2-COCO model on your local machine
2.   Upload it to your google drive
3.   Copy it to the notebook 
4.   Unzip it & move it inside models/research/object_detection folder 



In [0]:
!cp /mydrive/faster_rcnn_inception_v2_coco_2018_01_28.tar.gz ./

In [0]:
#uncompress a tar.gz in tensorflow1/research/object_detection directory
!tar -xvf faster_rcnn_inception_v2_coco_2018_01_28.tar.gz -C ./tensorflow1/models/research/object_detection

faster_rcnn_inception_v2_coco_2018_01_28/
faster_rcnn_inception_v2_coco_2018_01_28/model.ckpt.index
faster_rcnn_inception_v2_coco_2018_01_28/checkpoint
faster_rcnn_inception_v2_coco_2018_01_28/pipeline.config
faster_rcnn_inception_v2_coco_2018_01_28/model.ckpt.data-00000-of-00001
faster_rcnn_inception_v2_coco_2018_01_28/model.ckpt.meta
faster_rcnn_inception_v2_coco_2018_01_28/saved_model/
faster_rcnn_inception_v2_coco_2018_01_28/saved_model/saved_model.pb
faster_rcnn_inception_v2_coco_2018_01_28/saved_model/variables/
faster_rcnn_inception_v2_coco_2018_01_28/frozen_inference_graph.pb


In [0]:
#check if faster_rcnn_inception_v2_coco_2018_01_28 file exist now in ./tensorflow1/models/research/object_detection
!ls ./tensorflow1/models/research/object_detection

anchor_generators			  inputs_test.py
box_coders				  legacy
builders				  matchers
CONTRIBUTING.md				  meta_architectures
core					  metrics
data					  model_hparams.py
data_decoders				  model_lib.py
dataset_tools				  model_lib_test.py
dockerfiles				  model_lib_v2.py
eval_util.py				  model_lib_v2_test.py
eval_util_test.py			  model_main.py
exporter.py				  models
exporter_test.py			  model_tpu_main.py
export_inference_graph.py		  object_detection_tutorial.ipynb
export_tflite_ssd_graph_lib.py		  predictors
export_tflite_ssd_graph_lib_test.py	  protos
export_tflite_ssd_graph.py		  README.md
faster_rcnn_inception_v2_coco_2018_01_28  samples
g3doc					  test_data
inference				  test_images
__init__.py				  tpu_exporters
inputs.py				  utils


#Step 5: cloning training data scripts from my github repository

those scripts in the repository are written by: [Edje Electronics](https://github.com/EdjeElectronics) 
<br> the repository contains python scripts that use to generate the training data in step 6.it contains the images, annotation data, .csv files, and TFRecords needed to train the detector. It also contains Python scripts that are used to generate the training data. It has scripts to test out the object detection classifier on images.

In [0]:
%cd tensorflow1/models/research/object_detection

/content/tensorflow1/models/research/object_detection


In [0]:
!git clone https://github.com/galsaeedi/training-data-python-scripts-for-tensorflow-object-detection-API.git

Cloning into 'training-data-python-scripts-for-tensorflow-object-detection-API'...
remote: Enumerating objects: 23, done.
remote: Counting objects: 100% (23/23), done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 160 (delta 9), reused 0 (delta 0), pack-reused 137
Receiving objects: 100% (160/160), 283.67 MiB | 54.25 MiB/s, done.
Resolving deltas: 100% (68/68), done.
Checking out files: 100% (136/136), done.


In [0]:
%cd /content

/content


In [0]:
#extract all training-data-python-scripts-for-tensorflow-object-detection-API/* content into object_detection folder
#just to smplify things :)
!mv /content/tensorflow1/models/research/object_detection/training-data-python-scripts-for-tensorflow-object-detection-API/* /content/tensorflow1/models/research/object_detection/

I'm used these scripts to generate my own training data. If you want to practice **training your own electricity meter detector**, you can leave all the files as they are. And jump to step 7

**Otherwise**, If you want to train your own object detector, delete the following files (do not delete the folders):

* All files in \object_detection\images\train and \object_detection\images\test
* The “test_labels.csv” and “train_labels.csv” files in \object_detection\images
* All files in \object_detection\training **except labelmap.pbtxt file we will  just modify it later**
* All files in \object_detection\inference_graph

Now, you are ready to start from scratch in training your own object detector. This code will assume that all the files listed above were deleted, and will go on to explain how to generate the files for your own training dataset.



# step 6: Gathering and Labeling a Custom Dataset

OK.. take a deep breath! now you need to gathering your data and then labeling it.
<br>at this point,you must have replaced my train and test files here: " \object_detection\images\train and \object_detection\images\test" with yours 

If you want to label your image, **LabelImg** is a great tool for labeling images, and its GitHub page has very clear instructions on how to install and use it.
[LabelImg GitHub Link](https://github.com/tzutalin/labelImg)

<br>Well, Now after gathering and labeling your data. > your train and test file must contain images and xml files associated with each image and ready to covert it into scv





In [0]:
#create csv files from xml file 
#note: if you already have a csv file look like photo bellow so skip this command 
!python /content/tensorflow1/models/research/object_detection/xml_to_csv.py

Successfully converted xml to csv.
Successfully converted xml to csv.


Next, you have to **change some lines** in generate_tfrecord.py 

<br>open the generate_tfrecord.py file in a text editor. Replace the label map starting at line 31 with your own label map, where each object is assigned an ID number. This same number assignment will be used when configuring the labelmap.pbtxt file in the next step.

for example instead of my classes "counter , counter1 , counter2 ...etc". write yours 

    if row_label == 'counter':
        return 1
    elif row_label == 'counter1':
        return 2
    elif row_label == 'counter2':
        return 3
    elif row_label == 'counter3':
        return 4
    
    else:
        None


Then, generate the TFRecord files by issuing these commands 

In [0]:
!python /content/tensorflow1/models/research/object_detection/generate_tfrecord.py --csv_input=/content/tensorflow1/models/research/object_detection/images/train_labels.csv --image_dir=/content/tensorflow1/models/research/object_detection/images/train --output_path=/content/tensorflow1/models/research/object_detection/train.record
!python /content/tensorflow1/models/research/object_detection/generate_tfrecord.py --csv_input=/content/tensorflow1/models/research/object_detection/images/test_labels.csv --image_dir=/content/tensorflow1/models/research/object_detection/images/test --output_path=/content/tensorflow1/models/research/object_detection/test.record

Successfully created the TFRecords: /content/tensorflow1/models/research/object_detection/train.record
Successfully created the TFRecords: /content/tensorflow1/models/research/object_detection/test.record


Now, need to create a label map and edit the training configuration file.

<br> The label map tells the trainer what each object is by defining a mapping of class names to class ID numbers. open tenserflow1/models/research/object_detection/training/labelmap.pbtxt file in text editor  and replace it with yours 

```
item {
  id: 1
  name: 'WRITE YOUR FIRST CLASS'
}

item {
  id: 2
  name: 'SECOND CLASS'
}

item {
  id: 3
  name: 'THIRD'
}

item {
  id: 4
  name: 'FOURTH'
}

```

 and so on...

**The last thing to do before training** is changing some lines .config file
<br>the object detection training pipeline must be configured. It defines which model and what parameters will be used for training. This is the last step before running training!

1. Navigate to ./tensorflow1/models/research/object_detection/samples/configs and copy the faster_rcnn_inception_v2_pets.config file into the /object_detection/training directory. 

In [0]:
!mv /content/tensorflow1/models/research/object_detection/samples/configs/faster_rcnn_inception_v2_pets.config /content/tensorflow1/models/research/object_detection/training/

2. dowenload faster_rcnn_inception_v2_pets.config (i recommend just copy it to google drive and edit it there using txt editor)


In [0]:
#copy faster_rcnn_inception_v2_pets.config from here to my google drive 
!cp /content/tensorflow1/models/research/object_detection/training/faster_rcnn_inception_v2_pets.config /content/drive/'My Drive'/

In [0]:
!ls /content/mydrive/'My Drive'

In [0]:
!ls /mydrive

3. open the file with a text editor. There are several changes to make to the .config file, mainly changing the number of classes and examples, and adding the file paths to the training data.
<br>Make the following changes to the faster_rcnn_inception_v2_pets.config file. <br>**Note:** The paths must be entered with single forward slashes (NOT backslashes), or TensorFlow will give a file path error when trying to train the model! Also, the paths must be in double quotation marks ( " ), not single quotation marks ( ' ).


> 1. Line 9. Change num_classes to the number of different objects you want the classifier to detect. in mycase num_classes : 4 .

> 2. Line 106. Change fine_tune_checkpoint to: fine_tune_checkpoint : "/content/tensorflow1/models/research/object_detection/faster_rcnn_inception_v2_coco_2018_01_28/model.ckpt"

> 3. Lines 123 and 125. In the train_input_reader section, change input_path and label_map_path to:
*   input_path : "/content/tensorflow1/models/research/object_detection/train.record"
*   label_map_path: "/content/tensorflow1/models/research/object_detection/training/labelmap.pbtxt"

> 4. Line 130. Change num_examples to the number of images you have in the \images\test directory.

> 5. Lines 135 and 137. In the eval_input_reader section, change input_path and label_map_path to:
*   input_path : "/content/tensorflow1/models/research/object_detection/test.record"
*   label_map_path: "/content/tensorflow1/models/research/object_detection/training/labelmap.pbtxt"



Save the file after the changes have been made AND copy it again to train folder here on the notebook 


In [0]:
#copy it again after change it from drive to training folder 
!cp /mydrive/faster_rcnn_inception_v2_pets.config /content/tensorflow1/models/research/object_detection/training/faster_rcnn_inception_v2_pets.config

That’s it! The training job is all configured and ready to go!

#step 7: start training

compile the Protobuf files, which are used by TensorFlow to configure model and training parameters. Every .proto file in the \object_detection\protos directory must be called out individually by the command.

In [0]:
!protoc --python_out=./tensorflow1/models/research ./tensorflow1/models/research/object_detection/protos/anchor_generator.proto ./tensorflow1/models/research/object_detection/protos/argmax_matcher.proto ./tensorflow1/models/research/object_detection/protos/bipartite_matcher.proto ./tensorflow1/models/research/object_detection/protos/box_coder.proto ./tensorflow1/models/research/object_detection/protos/box_predictor.proto ./tensorflow1/models/research/object_detection/protos/eval.proto ./tensorflow1/models/research/object_detection/protos/faster_rcnn.proto ./tensorflow1/models/research/object_detection/protos/faster_rcnn_box_coder.proto ./tensorflow1/models/research/object_detection/protos/grid_anchor_generator.proto ./tensorflow1/models/research/object_detection/protos/hyperparams.proto ./tensorflow1/models/research/object_detection/protos/image_resizer.proto ./tensorflow1/models/research/object_detection/protos/input_reader.proto ./tensorflow1/models/research/object_detection/protos/losses.proto ./tensorflow1/models/research/object_detection/protos/matcher.proto ./tensorflow1/models/research/object_detection/protos/mean_stddev_box_coder.proto ./tensorflow1/models/research/object_detection/protos/model.proto ./tensorflow1/models/research/object_detection/protos/optimizer.proto ./tensorflow1/models/research/object_detection/protos/pipeline.proto ./tensorflow1/models/research/object_detection/protos/post_processing.proto ./tensorflow1/models/research/object_detection/protos/preprocessor.proto ./tensorflow1/models/research/object_detection/protos/region_similarity_calculator.proto ./tensorflow1/models/research/object_detection/protos/square_box_coder.proto ./tensorflow1/models/research/object_detection/protos/ssd.proto ./tensorflow1/models/research/object_detection/protos/ssd_anchor_generator.proto ./tensorflow1/models/research/object_detection/protos/string_int_label_map.proto ./tensorflow1/models/research/object_detection/protos/train.proto ./tensorflow1/models/research/object_detection/protos/keypoint_box_coder.proto ./tensorflow1/models/research/object_detection/protos/multiscale_anchor_generator.proto ./tensorflow1/models/research/object_detection/protos/graph_rewriter.proto ./tensorflow1/models/research/object_detection/protos/calibration.proto ./tensorflow1/models/research/object_detection/protos/flexible_grid_anchor_generator.proto

object_detection/protos/flexible_grid_anchor_generator.proto: File not found.
object_detection/protos/grid_anchor_generator.proto: File not found.
object_detection/protos/multiscale_anchor_generator.proto: File not found.
object_detection/protos/ssd_anchor_generator.proto: File not found.
tensorflow1/models/research/object_detection/protos/anchor_generator.proto: Import "object_detection/protos/flexible_grid_anchor_generator.proto" was not found or had errors.
tensorflow1/models/research/object_detection/protos/anchor_generator.proto: Import "object_detection/protos/grid_anchor_generator.proto" was not found or had errors.
tensorflow1/models/research/object_detection/protos/anchor_generator.proto: Import "object_detection/protos/multiscale_anchor_generator.proto" was not found or had errors.
tensorflow1/models/research/object_detection/protos/anchor_generator.proto: Import "object_detection/protos/ssd_anchor_generator.proto" was not found or had errors.
tensorflow1/models/research/obje

In [0]:
pip install -e ./tensorflow1/models/research/slim

Obtaining file:///content/tensorflow1/models/research/slim
  Attempting uninstall: slim
    Found existing installation: slim 0.1
    Can't uninstall 'slim'. No files were found to uninstall.
  Running setup.py develop for slim


In [0]:
!python ./tensorflow1/models/research/setup.py build

running build


In [0]:
!python ./tensorflow1/models/research/setup.py install

running install
running bdist_egg
running egg_info
creating object_detection.egg-info
writing object_detection.egg-info/PKG-INFO
writing dependency_links to object_detection.egg-info/dependency_links.txt
writing requirements to object_detection.egg-info/requires.txt
writing top-level names to object_detection.egg-info/top_level.txt
writing manifest file 'object_detection.egg-info/SOURCES.txt'
reading manifest file 'object_detection.egg-info/SOURCES.txt'
writing manifest file 'object_detection.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib

creating build
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying object_detection.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying object_detection.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying object_detection.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
co

move train.py from /object_detection/legacy into the /object_detection folder and then run the command below.

In [0]:
!mv ./tensorflow1/models/research/object_detection/legacy/train.py ./tensorflow1/models/research/object_detection/

In [0]:
#!python /content/tensorflow1/models/research/object_detection/train.py --logtostderr --train_dir=/content/tensorflow1/models/research/object_detection/training/ --pipeline_config_path=/content/tensorflow1/models/research/object_detection/training/faster_rcnn_inception_v2_pets.config

In [0]:
#training
!python /content/tensorflow1/models/research/object_detection/train.py --logtostderr --train_dir=/content/tensorflow1/models/research/object_detection/training/ --pipeline_config_path=/content/tensorflow1/models/research/object_detection/training/faster_rcnn_inception_v2_pets.config --label_map_path=/content/tensorflow1/models/research/object_detection/training/labelmap.pbtxt

The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
  * https://github.com/tensorflow/io (for I/O related ops)
If you depend on functionality not listed there, please file an issue.

Instructions for updating:
Use object_detection/model_main.py.
W0529 19:33:00.628545 140174439565184 deprecation.py:323] From /usr/local/lib/python3.6/dist-packages/absl/app.py:250: main (from __main__) is deprecated and will be removed in a future version.
Instructions for updating:
Use object_detection/model_main.py.

W0529 19:33:00.629017 140174439565184 module_wrapper.py:139] From /usr/local/lib/python3.6/dist-packages/object_detection/utils/config_util.py:96: The name tf.gfile.GFile is deprecated. Please use tf.io.gfile.GFile instead.

Instructions for updating:
Please switch to tf.train.create_global_step
W0529 19:33:00.63

Now that training is complete, the last step is to generate the frozen inference graph (.pb file). From the /object_detection/training folder, issue the following command, where “XXXX” in “model.ckpt-XXXX” should be replaced with the highest-numbered .ckpt file in the training folder:

In [0]:
!python ./tensorflow1/models/research/object_detection/export_inference_graph.py --input_type image_tensor --pipeline_config_path ./tensorflow1/models/research/object_detection/training/faster_rcnn_inception_v2_pets.config --trained_checkpoint_prefix ./tensorflow1/models/research/object_detection/training/model.ckpt-625 --output_directory ./tensorflow1/models/research/object_detection/inference_graph

The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
  * https://github.com/tensorflow/io (for I/O related ops)
If you depend on functionality not listed there, please file an issue.


W0529 19:11:11.888524 139951039092608 module_wrapper.py:139] From /usr/local/lib/python3.6/dist-packages/object_detection/exporter.py:357: The name tf.gfile.MakeDirs is deprecated. Please use tf.io.gfile.makedirs instead.


W0529 19:11:11.888848 139951039092608 module_wrapper.py:139] From /usr/local/lib/python3.6/dist-packages/object_detection/exporter.py:110: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

Instructions for updating:
Use `tf.cast` instead.
W0529 19:11:11.891833 139951039092608 deprecation.py:323] From /usr/local/lib/python3.6/dist-packages/object_detection/exporter.py:309: t

#step 8: test your detector :)

1. Upload your test image from your drive to the notebook.
2. open Object_detection_image.py change :
>*   IMAGE_NAME
>*   CWD_PATH(to odject_detection folder)
>*   NUM_CLASSES

**important**
3. UPDATE 29/5/20: tensorflow model authorizers updated their object_detection/utils/**label_map_util.py** file. So an error now appears when you try to use it with my notebook. Fortunatly,I have a copy of the old one. you have to upload it here from: https://drive.google.com/file/d/1DHl-Dp-38k8Hm4rSbZ2L0G8-J9k68CxJ/view?usp=sharing and replace it with the existing one **Otherwise you will get a kaypoint error**

In [0]:
#copy image from my drive to here
!cp /mydrive/meter1201.jpg ./

In [0]:
#copy label_map_util.py from my drive to here and replace it with the existing one 
!cp /mydrive/label_map_util.py ./tensorflow1/models/research/object_detection/utils/label_map_util.py

In [0]:
#test
!python /content/tensorflow1/models/research/object_detection/Object_detection_image.py




2020-05-29 19:50:43.329911: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2020-05-29 19:50:43.331671: E tensorflow/stream_executor/cuda/cuda_driver.cc:318] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
2020-05-29 19:50:43.331706: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (8dc09624c5a7): /proc/driver/nvidia/version does not exist
2020-05-29 19:50:43.338775: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2300000000 Hz
2020-05-29 19:50:43.340466: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x219d480 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-05-29 19:50:43.340519: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version


In [0]:
#!rm -rf /usr/local/lib/python3.6/dist-packages/tensorflow

In [0]:
#!ls /usr/local/lib/python3.6/dist-packages/tensorflow

ls: cannot access '/usr/local/lib/python3.6/dist-packages/tensorflow': No such file or directory


In [0]:
pip install utils